hedera-mirror-importer/src/main/resources/db/migration/v1/V1.95.0__add_gas_consumed_field.sql

```diff
@@ -0,0 +1,70 @@
+alter table if exists contract_result
+    add column if not exists gas_consumed bigint null;
+
+-- Function to get intrinsic gas for contract operation
+create or replace function get_intrinsic_gas(payload bytea, is_creation boolean) returns bigint as $$
+declare
+  zero_count bigint := 0;
+  non_zero_count bigint := 0;
+  i int;
+begin
+  for i in 1..octet_length(payload) loop
+    if get_byte(payload, i-1) = 0 then
+      zero_count := zero_count + 1;
+    else
+      non_zero_count := non_zero_count + 1;
+    end if;
+  end loop;
+  if is_creation then
+    return 21000 + 32000 + zero_count * 4 + non_zero_count * 16; -- contract creation cost calculation
+  else
+    return 21000 + zero_count * 4 + non_zero_count * 16; -- contract call cost calculation
+  end if;
+end;
+$$ language plpgsql immutable;
+
+-- Update contract_result with gas_consumed values.
+-- Use the contract_action with call_depth = 0, as this is the top-level action, and add intrinsic gas to it.
+with contract_action_gas_usage as (
+    select
+        consensus_timestamp,
+        gas_used
+    from
+        contract_action
+    where
+        call_depth = 0
+), contract_transaction_info as (
+    select
+        cr.consensus_timestamp,
+        cr.contract_id,
+        e.created_timestamp,
+        -- Correctly assign `payload` based on whether it's a contract creation or call.
+        case
+            when cr.consensus_timestamp = e.created_timestamp then c.initcode
+            else cr.function_parameters
+        end as payload,
+        -- Determine if this is a contract create or call transaction based on entity's created_timestamp
+        (cr.consensus_timestamp = e.created_timestamp) as is_creation
+    from
+        contract_result cr
+    left join contract c on cr.contract_id = c.id
+    left join entity e on c.id = e.id
+), gas_calculation as (
+    select
+        cti.consensus_timestamp,
+        cti.is_creation,
+        cti.payload,
+        cagu.gas_used
+    from
+        contract_transaction_info cti
+    -- Filter out only contract transactions with contract actions sidecars
+    join contract_action_gas_usage cagu on cti.consensus_timestamp = cagu.consensus_timestamp
+)
+update contract_result cr
+set gas_consumed = gc.gas_used +
+                   get_intrinsic_gas(gc.payload, gc.is_creation)
```

Comment:
sidecar is an optional feature in importer and mirrornode operators can choose to only process certain types of sidecars, so it's possible for a mirronode to process contract action sidecars but not bytecode, then `initcode` can be null.

```suggestion
                   get_intrinsic_gas(coalesce(gc.payload, ''::bytea), gc.is_creation)
```

Reply:
That's true. I've adjusted it, but in the case where the ContractBytecode sidecar does not exist, do we still want to populate the gas_consumed field for CREATE operations? As this would result in displaying an invalid gas_consumed value?

Reply:
We can leave it as is. It's the operator's choice to configure importer per the requirement and thus many things are expected to break as a result.

If we decide to convert the backfill part into a Java migration, we can check the importer configuration and skip it if contract action / contract bytecode sidecars are not enabled.